cassandraprotocol/frame/frame.go

```diff
@@ -110,43 +118,29 @@ func NewResponseFrame(
 	if !message.IsResponse() {
 		return nil, errors.New("message is not a response: opcode " + string(message.GetOpCode()))
 	}
-	return newFrame(
-		&Header{
+	return &Frame{
+		Header: &Header{
 			Version:          version,
 			StreamId:         streamId,
 			TracingRequested: tracingId != nil,
 		},
-		&Body{
+		Body: &Body{
 			TracingId:     tracingId,
 			CustomPayload: customPayload,
 			Warnings:      warnings,
 			Message:       message,
-		})
-}
-
-func newFrame(header *Header, body *Body) (*Frame, error) {
-	// Check header and body global conformity with protocol specs here.
-	// Body message conformity with protocol specs will be tested by the message codecs.
-	if header.Version < cassandraprotocol.ProtocolVersionMin || header.Version > cassandraprotocol.ProtocolVersionMax {
-		return nil, fmt.Errorf("unsupported protocol version: %v", header.Version)
-	}
-	if body.CustomPayload != nil && header.Version < cassandraprotocol.ProtocolVersion4 {
-		return nil, errors.New("custom payloads require protocol version 4 or higher")
-	}
-	if body.Warnings != nil && header.Version < cassandraprotocol.ProtocolVersion4 {
-		return nil, errors.New("warnings require protocol version 4 or higher")
-	}
-	return &Frame{header, body}, nil
+		},
+	}, nil
 }
 
-func (f *Frame) String() string {
+func (f Frame) String() string {
```

Comment:
Why this change for these 3 functions?

Reply:
Because I was having nil pointer panics here. I realized that when a receiver is a pointer, the method gets called event when the actual receiver is nil, as in:

```
var frame *Frame
println(frame.String())
```

The code above panics inside `String()`, _not_ outside of it.

Reply:
Are you sure that example works? You are passing a pointer but the function receives an actual value. To me the solution would be to check if the receiver is `nil`, no?

Reply:
No it doesn't work either, only your IDE now detects that you are going to have a nil pointer at the call site, not inside the method. But I will revert this change, it is not really related to this PR.

Reply:
Hmm well it's up to the caller then to make sure it is not calling methods/functions on `nils`